common-rest/src/main/java/org/uniprot/api/rest/search/AbstractSolrSortClause.java

```diff
@@ -18,34 +20,37 @@
  */
 public abstract class AbstractSolrSortClause {
     public static final String SCORE = "score";
+    protected List<Pair<String, Sort.Direction>> defaultFieldSortOrderPairs;
 
-    public Sort getSort(String sortClause, boolean hasScore) {
-        Sort result;
-        if (StringUtils.isEmpty(sortClause)) {
-            result = createDefaultSort(hasScore);
-        } else {
-            result = createSort(sortClause);
-        }
-
-        String documentIdFieldName = getSolrDocumentIdFieldName();
-        if (result != null && result.getOrderFor(documentIdFieldName) == null) {
-            result = result.and(new Sort(Sort.Direction.ASC, documentIdFieldName));
-        }
-        return result;
+    public Sort getSort(String sortClause) {
+        return StringUtils.isEmpty(sortClause) ? createDefaultSort() : createSort(sortClause);
     }
 
-    public Sort createSort(String sortClause) {
-        return convertToSolrSort(parseSortClause(sortClause));
-    }
-
-    protected abstract Sort createDefaultSort(boolean hasScore);
+    protected abstract List<Pair<String, Sort.Direction>> getDefaultFieldSortOrderPairs();
```

Comment:
It looks like all the implementations of this have more or less the same structure:
```java
        if (this.defaultFieldSortOrderPairs == null) {
            this.defaultFieldSortOrderPairs = new ArrayList<>();
            // add sorts to to this.defaultFieldSortOrderPairs
        }
        return this.defaultFieldSortOrderPairs;
```

If so, you could:
- initialise defaultFieldSortOrderPairs in AbstractSolrSortClause as a new ArrayList<>()
- remove the abstract defaultFieldSortOrderPairs method
- add a concrete method addDefaultFieldOrderPair(field, direction)
- in the implementation classes, call addDefaultFieldOrderPair in initialisation, to assert what the default fields are.

This would then extract only the differing code to the concrete classes, and would save more duplicated code.

Have I missed something? Seems like it'd work to me.